src/connectors/configure-related-items/connectConfigureRelatedItems.ts

```diff
@@ -0,0 +1,157 @@
+import algoliasearchHelper, {
+  SearchParameters,
+  PlainSearchParameters,
+} from 'algoliasearch-helper';
+import { Unmounter, WidgetFactory, ResultHit } from '../../types';
+import { createDocumentationMessageGenerator, warning } from '../../lib/utils';
+import connectConfigure, {
+  ConfigureRenderer,
+  ConfigureConnectorParams,
+} from '../configure/connectConfigure';
+
+export type MatchingPatterns = {
+  [attribute: string]: {
+    /**
+     * The score of the optional filter.
+     *
+     * @see https://www.algolia.com/doc/guides/managing-results/rules/merchandising-and-promoting/in-depth/optional-filters/
+     */
+    score: number;
+  };
+};
+
+export interface ConfigureRelatedItemsConnectorParams {
+  /**
+   * The reference hit to extract the filters from.
+   */
+  hit: ResultHit;
+  /**
+   * The schema to create the optional filters.
+   * Each key represents an attribute from the hit.
+   */
+  matchingPatterns: MatchingPatterns;
+  /**
+   * Function called to transform the search parameters generated.
+   */
+  transformSearchParameters?(
+    searchParameters: SearchParameters
+  ): PlainSearchParameters;
+}
+
+export type ConfigureRelatedItemsWidgetFactory<
+  TConfigureRelatedItemsWidgetParams
+> = WidgetFactory<
+  ConfigureRelatedItemsConnectorParams & TConfigureRelatedItemsWidgetParams
+>;
+
+type ConfigureRelatedItemsConnector = <TConfigureRelatedItemsWidgetParams>(
+  render?: ConfigureRenderer<ConfigureConnectorParams>,
+  unmount?: Unmounter
+) => ConfigureRelatedItemsWidgetFactory<TConfigureRelatedItemsWidgetParams>;
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'configure-related-items',
+  connector: true,
+});
+
+function createOptionalFilter({
+  attributeName,
+  attributeValue,
+  attributeScore,
+}) {
+  return `${attributeName}:${attributeValue}<score=${attributeScore || 1}>`;
+}
+
+const connectConfigureRelatedItems: ConfigureRelatedItemsConnector = (
+  renderFn,
+  unmountFn
+) => {
+  warning(
+    false,
```

Comment:
ideally you could opt-in to this widget without warning, maybe via an argument?

Reply:
IMHO it would be good to always warn about it in development. Users could set the argument, forget about it and then be surprised by a future breaking change in a minor version.

Reply:
Or name the connector `EXPERIMENTAL_connectConfigureRelatedItems`?

Reply:
Done in 8c29d7e4f8e1f6f8a9767a8e7c551b03970850ad.